src/metabase/pulse/render/body.clj

```diff
@@ -527,14 +528,16 @@
         "line"))
 
 (defn- join-series
-  [names colors types row-seqs y-axis-positions]
-  (vec (for [[card-name card-color card-type rows y-axis-position]
-             (map vector names colors types row-seqs y-axis-positions)]
-         {:name          card-name
-          :color         card-color
-          :type          card-type
-          :data          rows
-          :yAxisPosition y-axis-position})))
+  [names types row-seqs y-axis-positions y-col]
+  (for [[idx [card-name card-type rows y-axis-position]]
+        (map-indexed vector (map vector names types row-seqs y-axis-positions))]
```

Comment:
This is the kind of comment I expect to see from BE developers since I know my Clojure skill is still not that good, it's comparable to BE developers trying to make JS/TS works so things might work out but not in an idiomatic way.

Thanks a lot for the input, I think your suggestion is a lot cleaner 👍

Reply:
Fixed.